src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -551,7 +554,12 @@ public Task StartHostAsync(RHostStartupInfo startupInfo, IRSessionCallback callb
 
             var callback = _callback;
             if (!addToHistory && callback != null) {
-                return await callback.ReadUserInput(prompt, len, ct);
+                try {
+                    IsReadingUserInput = true;
+                    return await callback.ReadUserInput(prompt, len, ct);
```

Comment:
the callback ends in `interactiveWindows.ReadStandardInput()` which is not cancellable AFAICT.